### DISCUSSION TITLE
Does Vendor actually support simple-json?

### DISCUSSION THREAD

**Discussion by meatandfat** [vendurehq/vendure#2949] [Q&A] [answered]
Hi.

On https://docs.vendure.io/guides/developer-guide/custom-fields/#list , it says, “For lists of primitive types (anything except relation), the database type will be set to simple-json which serializes the data into a JSON string.”

And according to https://orkhan.gitbook.io/typeorm/docs/entities#simple-json-column-type , I should be able to create a simple-json column directly in my custom entity.

I’ve tried many things. In a custom entity,

```
    @Column("int", { array: true })
    one_name: number[];

    // @Column()
    // code: string[] = [];
    @Column("simple-array")
    new_name: string[];

    @Column("simple-array")
    names: string[]

    @Column("simple-json")
    profile: { name: string; nickname: string }

    @Column("simple-json")
    profile2: { name: int; nickname: int }

    @Column('simple-json', {nullable: true})
    otherInfo: {
        description: string;
        freeShipping: boolean;
        weight: number;
        color: string;
    };
```

And in vendure-config.ts,
```
customFields: {
		Product: [

            {
                name: 'infoUrls',
                type: 'simple-json',
                list: true,
                madeUpKey: "hi"
            },
		],
```

Of these, all have their type set as “text” except for one_name: number[] which is set as I expected.

**Comment by michaelbromley** [comment, marked as answer]
Hi,

The "simple-json" type only exists internally in TypeORM - as far as the database is concerned, it's just a string of text. TypeORM is responsible for serializing & deserializing the data into the expected object.

In terms of custom fields (like your `infoUrls` example), the `type: 'simple-json'` is not valid in Vendure. The permitted custom field types are listed here: https://docs.vendure.io/guides/developer-guide/custom-fields/#available-custom-field-types

This line:

> For lists of primitive types (anything except relation), the database type will be set to simple-json which serializes the data into a JSON string

just means that internally, if you set a custom field to e.g. `{ type: 'string', list: true }`, Vendure will make use of the TypeORM "simple-json" type to handle the serializing/deserializing into text when storing in the database.

**Reply by meatandfat** [reply]
Interesting. Looking at the Typeorm page again, I see it can work with databases without a json type. Since some do, I think their 

> Will be stored in a single database column as {"name":"John","nickname":"Malkovich"} value. When you'll load data from the database, you will have your object/array/primitive back via JSON.parse

is a little misleading.

.
> In terms of custom fields (like your infoUrls example), the type: 'simple-json' is not valid in Vendure.
 
Yes! An artifact from trying anything I could thing of 😀

Thanks for the help!
